### PR TITLE
Plugin Initialization and PSR4 Structure

### LINKED ISSUE #1: Create PSR4 Plugin Structure
Initialize plugin in PSR4 structure with composer, JS linting, and PHP coding standards.

- [x] PSR4
- [x] ESLINT
- [x] PHPCS

### PR DESCRIPTION
**Pull request by ronalfy**
Resolves #1 

Adds plugin headers and wrapper for `plugins_loaded` and `init` actions.

This PR serves to act as the foundation for the plugin structure.

### CODE REVIEW

File: class-phpdoc-plugin.php [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+ * Requires at least: 5.8
+ * Text Domain: phpdoc-plugin
+ *
+ * @package phpdocplugin
+ */
+
+namespace Pods\phpdocplugin;
+
+define( 'PHPDOCPLUGIN_FILE', __FILE__ );
+define( 'PHPDOCPLUGIN_VERSION', '0.0.1' );
```
Comment (sc0ttkclark) [inline class-phpdoc-plugin.php]:
```suggestion
define( 'PHPDOC_PLUGIN_FILE', __FILE__ );
define( 'PHPDOC_PLUGIN_VERSION', '0.0.1' );
```

File: composer.json [hunk 2]
```diff
@@ -0,0 +1,34 @@
+{
+  "name": "pods-framework/phpdocplugin",
```
Comment (sc0ttkclark) [inline composer.json]:
```suggestion
  "name": "pods-framework/phpdoc-plugin",
```

File: .gitignore [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+composer.lock
+/node_modules/
+vendor/*
+npm-debug.log
+
+# PHPunit config
+phpunit.xml
+
+# Source maps should just be built locally
+ui/js/pods-ui-ready.min.js.map
```
Comment (sc0ttkclark) [inline .gitignore]:
```suggestion
```

File: class-phpdoc-plugin.php [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+ * Plugin Name: phpDoc Plugin
+ * Plugin URI: https://github.com/pods-framework/phpdoc-plugin
+ * Description: A companion plugin for docs parsing with the PHPDoc Parser.
+ * Author: Pods Framework Team
+ * Author URI: https://pods.io/about/
+ * Version: 0.0.1
+ * Requires at least: 5.8
+ * Text Domain: phpdoc-plugin
+ *
+ * @package phpdocplugin
```
Comment (sc0ttkclark) [inline class-phpdoc-plugin.php]:
```suggestion
 * @package Pods\phpDoc_Plugin
```

File: class-phpdoc-plugin.php [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+ */
+
+namespace Pods\phpdocplugin;
```
Comment (sc0ttkclark) [inline class-phpdoc-plugin.php]:
```suggestion
namespace Pods\phpDoc_Plugin;
```

File: class-phpdoc-plugin.php [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+
+// Support for site-level autoloading.
+if ( file_exists( __DIR__ . '/vendor/autoload.php' ) ) {
+	require_once __DIR__ . '/vendor/autoload.php';
+}
+
+/**
+ * Main plugin class.
+ */
+class PHPDoc_Plugin {
```
Comment (sc0ttkclark) [inline class-phpdoc-plugin.php]:
```suggestion
class Bootstrap {
```

File: class-phpdoc-plugin.php [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+	 */
+	public function plugins_loaded() {
+		$plugin = new Plugin();
+
+		// Init the plugin.
+		add_action( 'init', array( $plugin, 'init' ) );
+	}
+}
+
+$phpdoc_plugin = new PHPDoc_Plugin();
```
Comment (sc0ttkclark) [inline class-phpdoc-plugin.php]:
```suggestion
$phpdoc_plugin = new Bootstrap();
```

File: composer.json [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+      "role": "Lead Developer"
+    }
+  ],
+  "require": {
+    "php": ">=7.4",
+    "composer/installers": "^1.7"
+  },
+  "autoload": {
+    "psr-4": {
+      "Pods\\phpdocplugin\\": "php/"
```
Comment (sc0ttkclark) [inline composer.json]:
```suggestion
      "Pods\\phpDoc_Plugin\\": "php/"
```

File: composer.json [hunk 9]
```diff
@@ -0,0 +1,34 @@
...
+    }
+  },
+  "autoload-dev": {
+    "psr-4": {
+      "Pods\\phpdocplugin\\": "tests/php/"
```
Comment (sc0ttkclark) [inline composer.json]:
```suggestion
      "Pods\\phpDoc_Plugin\\": "tests/php/"
```

File: php/Plugin.php [hunk 10]
```diff
@@ -0,0 +1,264 @@
+<?php
+/**
+ * Plugin Utility class.
+ *
+ * @package phpdocplugin
+ */
+
+namespace Pods\phpdocplugin;
```
Comment (sc0ttkclark) [inline php/Plugin.php]:
```suggestion
namespace Pods\phpDoc_Plugin;
```

File: php/Plugin.php [hunk 11]
```diff
@@ -0,0 +1,264 @@
+<?php
+/**
+ * Plugin Utility class.
+ *
+ * @package phpdocplugin
```
Comment (sc0ttkclark) [inline php/Plugin.php]:
```suggestion
 * @package Pods\phpDoc_Plugin
```

File: php/Plugin.php [hunk 12]
```diff
@@ -0,0 +1,264 @@
...
+	 *
+	 * @var string
+	 */
+	protected $dir;
+
+	/**
+	 * Setup the plugin.
+	 */
+	public function __construct() {
+		$this->file = PHPDOCPLUGIN_FILE;
```
Comment (sc0ttkclark) [inline php/Plugin.php]:
```suggestion
		$this->file = PHPDOC_PLUGIN_FILE;
```

File: php/Plugin.php [hunk 13]
```diff
@@ -0,0 +1,264 @@
...
+		$this->dir  = dirname( PHPDOCPLUGIN_FILE );
```
Comment (sc0ttkclark) [inline php/Plugin.php]:
```suggestion
		$this->dir  = dirname( PHPDOC_PLUGIN_FILE );
```

File: php/Plugin.php [hunk 14]
```diff
@@ -0,0 +1,264 @@
...
+		return $meta;
+	}
+
+	/**
+	 * Return the plugin slug.
+	 *
+	 * @return string plugin slug.
+	 */
+	public static function get_plugin_slug() {
+		return dirname( plugin_basename( PHPDOCPLUGIN_FILE ) );
```
Comment (sc0ttkclark) [inline php/Plugin.php]:
```suggestion
		return dirname( plugin_basename( PHPDOC_PLUGIN_FILE ) );
```

File: php/Plugin.php [hunk 15]
```diff
@@ -0,0 +1,264 @@
...
+	}
+
+	/**
+	 * Return the basefile for the plugin.
+	 *
+	 * @return string base file for the plugin.
+	 */
+	public static function get_plugin_file() {
+		return plugin_basename( PHPDOCPLUGIN_FILE );
```
Comment (sc0ttkclark) [inline php/Plugin.php]:
```suggestion
		return plugin_basename( PHPDOC_PLUGIN_FILE );
```

File: php/Plugin.php [hunk 16]
```diff
@@ -0,0 +1,264 @@
...
+	}
+
+	/**
+	 * Return the version for the plugin.
+	 *
+	 * @return float version for the plugin.
+	 */
+	public static function get_plugin_version() {
+		return PHPDOCPLUGIN_VERSION;
```
Comment (sc0ttkclark) [inline php/Plugin.php]:
```suggestion
		return PHPDOC_PLUGIN_VERSION;
```

File: php/Plugin.php [hunk 17]
```diff
@@ -0,0 +1,264 @@
...
+
+	/**
+	 * Get the plugin directory for a path.
+	 *
+	 * @param string $path The path to the file.
+	 *
+	 * @return string The new path.
+	 */
+	public static function get_plugin_dir( $path = '' ) {
+		$dir = rtrim( plugin_dir_path( PHPDOCPLUGIN_FILE ), '/' );
```
Comment (sc0ttkclark) [inline php/Plugin.php]:
```suggestion
		$dir = rtrim( plugin_dir_path( PHPDOC_PLUGIN_FILE ), '/' );
```

File: php/Plugin.php [hunk 18]
```diff
@@ -0,0 +1,264 @@
...
+
+	/**
+	 * Return a plugin URL path.
+	 *
+	 * @param string $path Path to the file.
+	 *
+	 * @return string URL to to the file.
+	 */
+	public static function get_plugin_url( $path = '' ) {
+		$dir = rtrim( plugin_dir_url( PHPDOCPLUGIN_FILE ), '/' );
```
Comment (sc0ttkclark) [inline php/Plugin.php]:
```suggestion
		$dir = rtrim( plugin_dir_url( PHPDOC_PLUGIN_FILE ), '/' );
```

File: php/Plugin.php [hunk 19]
```diff
@@ -0,0 +1,264 @@
...
+	public static function get_plugin_version() {
+		return PHPDOCPLUGIN_VERSION;
+	}
+
+	/**
+	 * Returns appropriate html for KSES.
+	 *
+	 * @param bool $svg Whether to add SVG data to KSES.
+	 */
+	public static function get_kses_allowed_html( $svg = true ) {
```
Comment (sc0ttkclark) [inline php/Plugin.php]: Is this necessary?

File: php/Plugin.php [hunk 20]
```diff
@@ -0,0 +1,264 @@
...
+	}
+
+	/**
+	 * Get plugin meta data.
+	 *
+	 * @param  string $field Optional field key.
+	 *
+	 * @return array|string|null
+	 */
+	public function meta( $field = null ) {
```
Comment (sc0ttkclark) [inline php/Plugin.php]: Is this necessary?

File: php/Plugin.php [hunk 21]
```diff
@@ -0,0 +1,264 @@
...
+		return ( defined( 'SCRIPT_DEBUG' ) && SCRIPT_DEBUG );
+	}
+
+	/**
+	 * Return the current version of the plugin.
+	 *
+	 * @return mixed
+	 */
+	public function version() {
+		return $this->meta( 'Version' );
```
Comment (sc0ttkclark) [inline php/Plugin.php]: There's a static function get_plugin_version() that returns the constant, is this method necessary?

File: php/Plugin.php (hunk 19 above)
Comment (ronalfy) [reply php/Plugin.php]: No, just filler from a plugin I wrote. I'll remove it.

File: php/Plugin.php (hunk 20 above)
Comment (ronalfy) [reply php/Plugin.php]: This is a nice to have, but I'll remove.

File: php/Plugin.php (hunk 21 above)
Comment (ronalfy) [reply php/Plugin.php]: yes, this is redundant. Thanks.

**Pull request merged**